fac/visual_regression_predictor
# refactor peter

- Initial commit, basic gem with some option parsing, guard, and rubocop
- Support passing in a file as the source of the changeset
- WIP: CLI parser working
- Adding artifacts
- Pass the artifacts_dir to mapped_project_files
- Testing for the artifacts dir flag
- Add project mapper and top-level method call
- Move top-level method call for CliOptionParser to the top of the file
- Adding tests for artifacts flag
- Tidying up and adding BlockList specs
- Tidying up ProjectMapping and specs
- Applying rubocop fixes
- Set a default artifacts directory within ProjectMapping
- Replace CliOptionParser with CliOptions, and updated specs
- Prevent guard from triggering VisualRegressionPredictor.run
- Small tidy up
- Rubocop fixes
- Adding ChangedFiles module and specs
- Rubocop fixes
- WIP: Adding GitHubClient class, and specs
- WIP: Rough approximation of running branch comparison or daily summary
- Adding a Logger class and specs
- Applying rubocop fixes
- WIP: split out determining matches into it's own class
- WIP: split out finding associated PRs to it's own class
- WIP: Splitting out the different modes, and writing output message to a file!
- Adding in branch mode flag to differentiate between a summary run and a branch run
- Prevent rspec and guard from triggering the run invocation on require
- Providing a nicer PR message, and updating the workflow
- Updating Gemfile.lock
- Deleting old reports directory
- Tidying up artifacts directory
- Quickly mock root_path within spec for now
- Removing old script files
- Get rubocop passing and tidying up old files
- Provide a working dir for the checkout
- Re-add support for passing in a ref
- Use the correct flag: --full_repo_name
- Type conversion on the branch_run_mode check
- More suggestions to the PR comment
- Updating GitHub formatted message and README
- Split running predictor from command line from code
- move logic to runner and simplify
